Why is EllipseF a class?

I stumbled upon `EllipseF` unlike the other shapes this is a `class` and not a `struct`.

It also does not implement `IShapeF`.

I've converted the class to a struct, added some DataContract attributes (similar to CircleF) and added various overloads and overrides, so the struct is inline with CircleF.

However its failing the windows build step, in the Tiled project (via GitHub)
MonoGame.Extended.Tiled.csproj : error NU1102: Unable to find package NETStandard.Library with version (>= 2.0.3)

--- a/src/dotnet/MonoGame.Extended/Math/EllipseF.cs
+++ b/src/dotnet/MonoGame.Extended/Math/EllipseF.cs
@@ -67,6 +67,27 @@ namespace MonoGame.Extended
                    && ellispse.RadiusY == RadiusY;
         }
 
+        public override bool Equals(object obj)
+        {
+            return obj is EllipseF && Equals((EllipseF)obj);
+        }
+
+        public override int GetHashCode()
+        {
+            unchecked
+            {
+                var hashCode = Center.GetHashCode();
+                hashCode = (hashCode * 397) ^ RadiusX.GetHashCode();
+                hashCode = (hashCode * 397) ^ RadiusY.GetHashCode();
+                return hashCode;
+            }
+        }
+
+        public override string ToString()
+        {
+            return $"Centre: {Center}, RadiusX: {RadiusX}, RadiusY: {RadiusY}";
+        }
+
         public static bool operator ==(EllipseF first, EllipseF second)
         {
             return first.Equals(ref second);

--- a/src/dotnet/MonoGame.Extended/Math/EllipseF.cs
+++ b/src/dotnet/MonoGame.Extended/Math/EllipseF.cs
@@ -1,20 +1,29 @@
 ﻿using System;
+using System.Runtime.Serialization;
 using Microsoft.Xna.Framework;
 
 namespace MonoGame.Extended
 {
-    public class EllipseF
+    [DataContract]
+    public struct EllipseF : IEquatable<EllipseF>, IEquatableByRef<EllipseF>, IShapeF
     {
+        [DataMember] public Vector2 Center { get; set; }
+        [DataMember] public float RadiusX { get; set; }
+        [DataMember] public float RadiusY { get; set; }
+
+        public Point2 Position
+        {
+            get => Center;
+            set => Center = value;
+        }
+
         public EllipseF(Vector2 center, float radiusX, float radiusY)
         {
             Center = center;
             RadiusX = radiusX;
             RadiusY = radiusY;
         }
-
-        public Vector2 Center { get; set; }
-        public float RadiusX { get; set; }
-        public float RadiusY { get; set; }
+    
         public float Left => Center.X - RadiusX;
         public float Top => Center.Y - RadiusY;
         public float Right => Center.X + RadiusX;
@@ -43,6 +52,29 @@ namespace MonoGame.Extended
         public bool Contains(Vector2 point)
         {
             return Contains(point.X, point.Y);
+        }
+
+        public bool Equals(EllipseF ellispse)
+        {
+            return Equals(ref ellispse);
+        }
+
+        public bool Equals(ref EllipseF ellispse)
+        {
+            // ReSharper disable once CompareOfFloatsByEqualityOperator
+            return ellispse.Center == Center 
+                   && ellispse.RadiusX == RadiusX 
+                   && ellispse.RadiusY == RadiusY;
+        }
+
+        public static bool operator ==(EllipseF first, EllipseF second)
+        {
+            return first.Equals(ref second);
+        }
+
+        public static bool operator !=(EllipseF first, EllipseF second)
+        {
+            return !(first == second);
         }
     }
 }

